test/unit-tests/function/statistics/correlation.test.js

```diff
@@ -0,0 +1,18 @@
+import assert from 'assert'
+import math from '../../../../src/defaultInstance.js'
+const corr = math.corr
+
+describe('correlation', function () {
+  it('should return the correlation coefficient from an array', function () {
+    assert.strictEqual(corr([1, 2, 3, 4, 5], [4, 5, 6, 7, 8]), 1)
```

Comment:
Can you also implement support for BigNumbers? It would mostly mean replacing operators `acc + x` with mathjs functions like `addScalar(acc, x)` that you can inject as a dependency. Or you could use use mathjs functions `sum` and `pow` right away.

If to complicated for now we could leave that for an other PR.

Reply:
So I mean: use mathjs functions instead of operators so the function "magically" works with numbers, BigNumbers, Fraction.

Reply:
Support for BigNumber is also added. Please check.

Reply:
Cool 😎